src/withAsyncHandlers.js

```diff
@@ -1,24 +1,46 @@
 import React, { Component } from 'react';
-import { forEachObjIndexed} from 'ramda';
+import { forEachObjIndexed, pathOr, pick, prop, when } from 'ramda';
+import { toSuccess, toError, toRequest } from './actionHelpers';
+import { takeEvery } from 'redux-saga/effects';
+import { withSagas } from './withSagas';
 
-export const withAsyncHandlers = asyncActionsHandlers => {
+const handlerTakers = {
+    successHandler: toSuccess,
+    errorHandler: toError,
+    pendingHandler: toRequest,
+};
+
+export const withAsyncHandlers = actions => {
     return WrappedComponent => class extends Component {
-        componentWillReceiveProps(nextProps) {
-            forEachObjIndexed((handlers, key) => {
-                if (this.props[key] && this.props[key].meta) {
-                    const { success : successHandler, error : errorHandler, pending : pendingHandler } = handlers;
-                    const { success, error, pending } = this.props[key].meta;
-                    const { meta } = nextProps[key];
+        constructor(props) {
+            super(props);
+            const sagas = [];
+
+            forEachObjIndexed(when(prop('saga'), (action) => {
```

Comment:
seems like you no need index here

Reply:
const sagas = map(...)

Reply:
moved to next `forEachObjIndexed` for only one actions handler